File: API/Client/Pagination.js
goog.provide('API.Client.Pagination');

/**
 * @record
 */
API.Client.Pagination = function() {}

/**
 * Current index page returned
 * @type {!number}
 * @export
 */
API.Client.Pagination.prototype.currentPage;

/**
 * Number of items per page returned
 * @type {!number}
 * @export
 */
API.Client.Pagination.prototype.size;

/**
 * Total number of pages available
 * @type {!number}
 * @export
 */
API.Client.Pagination.prototype.totalPages;

/**
 * Total number of elements (items) available
 * @type {!number}
 * @export
 */
API.Client.Pagination.prototype.totalElements;

/**
 * @type {!Array<!API.Client.Sort>}
 * @export
 */
API.Client.Pagination.prototype.sort;

File: client/scalaz/src/main/scala/org/openapitools/client/api/Pagination.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime


import Pagination._

case class Pagination (
  /* Current index page returned */
  currentPage: Option[Integer],
/* Number of items per page returned */
  size: Option[Integer],
/* Total number of pages available */
  totalPages: Option[Integer],
/* Total number of elements (items) available */
  totalElements: Option[Integer],
sort: Option[List[Sort]])

object Pagination {
  import DateTimeCodecs._

  implicit val PaginationCodecJson: CodecJson[Pagination] = CodecJson.derive[Pagination]
  implicit val PaginationDecoder: EntityDecoder[Pagination] = jsonOf[Pagination]
  implicit val PaginationEncoder: EntityEncoder[Pagination] = jsonEncoderOf[Pagination]
}
